File: py-ex/examples/ex_1.py
from helpers.shared import getJsonPath, getDataFromKey, jsonObject2IntVector

def ex_1( key: str ):
    
    jsonPath = getJsonPath( 1 )
    jsonData = getDataFromKey( jsonPath, key )
    
    if jsonData == None:
        print(f'Error Reading JSON File. Data with Key-${key} not found')
        return
    
    if jsonData == None: return
    
    data = jsonObject2IntVector( jsonData )

    if data == None: 
        print(f'Fail Conversion Data')
        return
    
    gifts = prepare_gifts( data )    
    print( gifts )

    
    
def prepare_gifts(gifts: list[int]) -> list[int]:
  return sorted( list( set( gifts ) ) )

File: node-ex/src/examples/ex_1.ts
import {
  getJsonPath,
  getDataFromKey,
  jsonObject2IntVector,
  IntVect,
  CustomNumberArray
} from "../utils";

export async function ex_1(key: string) {
  const jsonPath = getJsonPath(1);
  const jsonData = await getDataFromKey(jsonPath, key);

  if (jsonData === null) {
    console.error("Error Parsing JSON");
    return;
  }

  const data: number[] | null = jsonObject2IntVector(jsonData);

  if (data === null) {
    console.error("Error Loading Data");
    return;
  }

  const gifts = prepareGifts(data);
  console.log(gifts);
}

function prepareGifts(gifts: IntVect): IntVect {
//   return [...new Set(gifts)].sort( (a,b) => a-b );
  return new CustomNumberArray( ...new Set(gifts) ).sortBy('ascending');
}
